.claude/scripts/pre-commit-validation.sh

```
@@ -413,6 +413,59 @@ if [[ -n "$PY_FILES" ]]; then
+# These checks prevent naive datetime patterns from reaching main.
+
+STAGED_PY_FILES=$(git diff --cached --name-only --diff-filter=ACM | grep '\.py$' || true)
+
+if [ -n "$STAGED_PY_FILES" ]; then
+  echo "🕐 Datetime timezone safety checks..."
+
+  # Check 1: Naive datetime.now() detection
+  NAIVE_NOW=$(echo "$STAGED_PY_FILES" | while IFS= read -r f; do [ -n "$f" ] && grep -Hn 'datetime\.now()' "$f" 2>/dev/null; done | grep -v 'now(UTC\|now(timezone\|now(tz=' || true)
+  if [ -n "$NAIVE_NOW" ]; then
+    echo "  ❌ Found naive datetime.now() — use datetime.now(UTC) instead:"
+    echo "$NAIVE_NOW" | sed 's/^/    /'
```

Shellcheck (0.11.0)

[style] See if you can use ${variable//search/replace} instead.

(SC2001)